pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Container Contracts
+excerpt: A guide to using container contracts to enforce device compatibility
+---
+
+# Container contracts
+
+__Note__ Container contracts are available for devices running Supervisor versions >= 10.6.0. All prior versions will not enforce any contracts.
+
+Container contracts are used to ensure the compatibility of a device to run an application service. For example, container contracts may be used to ensure that the host operating system is running a specific version of balenaOS or Supervisor such that if an application is dependent on a specific version, it only gets deployed to those devices that can meet these requirements to fulfill the contract. Currently, you may define contract requirements based on the versions of [balenaOS](/reference/OS/overview/2.x/) and [Supervisor](/reference/supervisor/supervisor-api/).
+
+Each service container can define the contracts that it enforces. If a container's contract requirements are not met, the release is not deployed to the device. You only need to define a contract for service containers that you wish to enforce a contract.
+
+## Contract specification
+
+Container contracts are defined as a `contract.yml` file and must be placed in the application structure alongside the corresponding service Dockerfile. For example, the following [multi-container application](/learn/develop/multicontainer/) defines two services named `first-service` and `second-service`, each with a corresponding `contract.yml` file.
```

Comment:
This is a little strange, as the Dockerfile is not necesarilly required, for example Dockerfile.template, and also package.json are supported. I'd say maybe "the root of the build context for a service"

Reply:
Thanks, that's just my lack of understanding on that one 😄